lib/Fetcher/FeedFetcher.php

```diff
@@ -321,10 +321,29 @@ protected function buildFeed(FeedInterface $feed, string $url, bool $getFavicon,
         }
         $newFeed->setAdded($this->time->getTime());
 
-        if (!$getFavicon) {
-            return $newFeed;
+        $favicon = $feed->getLogo();
+
+        // check if feed has a logo
+        if (!is_null($favicon)) {
+            $favicon_path = join("/", $this->ITempManager->getTempBaseDir(), basename(($favicon)));
+            file_put_contents(
+                $favicon_path,
+                file_get_contents($url));
+            $is_image = substr(mime_content_type($favicon_path), 0, 5) === "image";
+            // check if file is actually an image
+            if($is_image){
+                list($width, $height, $type, $attr) = getimagesize($favicon_path);
+                // check if image is square else fall back to favicon
+                if($width !== $height){
+                    $favicon = $this->faviconFactory->get($url);
+                }
+            }
+        }else{
+            $favicon = $this->faviconFactory->get($url);
         }
+
         $favicon = $this->faviconFactory->get($url);
```

Comment:
probably a merging mistake as this would fetch the favicon anyway, no matter what happened before.

(and missing spaces around `else` in L341)

Reply:
yea copy paste mistake, thanks also fixed the other spacing issues